07-clustering.Rmd

```diff
@@ -0,0 +1,124 @@
+
+## Calculating and choosing PCs
+
+We first use principal component analysis (PCA), which is a dimensionality reduction method that maximizes the amount of variation captured by each component, or PC. 
+
+It's up to the researcher to choose how many PCs to use for downstream analyses. More PCs mean that more biological signal is retained in the data, but it also increases the potential for noise. In our analyses, we will use 50 (the default).
+
+We are also going to examine how the number of genes we choose to keep in our data cleaning step affects the PC choice.
+
+```{r, warning = FALSE, message = FALSE}
+library(scran)
+
+set.seed(100)
+sce.zeisel.PCA_500 <- fixedPCA(sce.zeisel.qc, subset.row=top500.hvgs)
+sce.zeisel.PCA.1000 <- fixedPCA(sce.zeisel.qc, subset.row=top1000.hvgs)
+sce.zeisel.PCA.2000 <- fixedPCA(sce.zeisel.qc, subset.row=top2000.hvgs)
+
+#looking at the variance explained by PCs when keeping different numbers of top genes
+percent.var500 <- attr(reducedDim(sce.zeisel.PCA.500), "percentVar")
+plot(percent.var500, log="y", xlab="PC", ylab="Variance explained (%)", main="Top 500 genes")
+
+percent.var1000 <- attr(reducedDim(sce.zeisel.PCA.1000), "percentVar")
+plot(percent.var1000, log="y", xlab="PC", ylab="Variance explained (%)", main="Top 1000 genes")
+
+percent.var2000 <- attr(reducedDim(sce.zeisel.PCA.2000), "percentVar")
+plot(percent.var2000, log="y", xlab="PC", ylab="Variance explained (%)", main="Top 2000 genes")
+```
+
+
+::: {.fyi}
+QUESTION
+1. How does including more genes affect the percentage of variation explained by the first few PCs?
+:::
+
+## Applying non-linear visualization methods to PCs
+
+In scRNA-seq analysis, plotting PCs generally does not offer enough resolution to visualize cell clusters. Instead, we rely on additional dimensionality reduction methods that can use non-linear data transformation algorithms. The most common approach is the t-stochastic neighbor embedding (t-SNE) method (Van der Maaten and Hinton 2008). t-SNE maps high-dimensional data in a low-dimensional space in such a way that if two points are close to each other in the final t-SNE plot, those two points were also close to each other in the original high-dimensional plot. (The same does not necessarily hold true for two points that are distant from each other; t-SNE only preserves the relative locations of neighboring clusters and points.)
+
+The t-SNE approach is computationally complex. In practice, we reduce the computational complexity in scRNA-seq analysis by performing t-SNE calculations on the top PCs in a dataset. We also need to set an initial starting seed and the perplexity parameter. This parameter  will determine the resolution of the plot. Lower perplexity values allow for finer resolution of population structure but can also be noisy.  It's a good idea to test multiple perplexity values when running your t-SNE analysis.
```

Comment:
this is getting fairly complex. I wonder if it's worth having some pictures to break things up? Conceptual diagrams perhaps?

Reply:
I rewrote this section - hopefully it's less dense now